src/relations/async_replication.py

```diff
@@ -487,7 +487,7 @@ def _on_async_relation_changed(self, event: RelationChangedEvent) -> None:
     def _on_async_relation_departed(self, event: RelationDepartedEvent) -> None:
         """Set a flag to avoid setting a wrong status message on relation broken event handler."""
         # This is needed because of https://bugs.launchpad.net/juju/+bug/1979811.
-        if event.departing_unit == self.charm.unit:
+        if event.departing_unit == self.charm.unit or self.charm._peers is None:
```

Comment:
This additional check was previously done in the `postgresql_client` interface relations. It's needed because sometimes the peer relation is not available anymore and we need to prevent destructive actions in the `relation-broken` hook when scaling down the cluster.

Reply:
Shouldn't this be `self.charm._peers is not None`? If `self.charm._peers` is None, it won't have `data` attr to update.

Reply:
You're right, Dragomir!